bsip-0078.md

```diff
@@ -0,0 +1,66 @@
+```
+BSIP: 77
+Title: ed25519-sha-256 algorithm addition to HTLC 
+Authors: ioBanker <Ammar.M@ioBanker.com>
+Status: Draft
+Type: Protocol
+Created: 2019-10-03
+Discussion: https://github.com/bitshares/bitshares-core/issues/2018
+```
+
+Abstract
+===================
+This BSIP proposes additional feature to Hashed Time Lock Contracts (HTLC) defined in [BSIP 44](https://github.com/bitshares/bsips/blob/master/bsip-0044.md). It adds the ed25519-sha-256 algorithm to the set of allowed hash functions.
+
+Motivation
+=================
+BitShares is aiming to become bridge of assets for market makers providing cross-currency liquidity as a software to individuals and corporate, supporting the ed25519-sha-256 pre-image hashing algorithm would enable BitShares to become compatible with the financial system transmission and it's future functions.
+
+Today banks are deploying solutions for Cross-Currency Settlement and FX Market Making.
+
+When those solutions are ILP-enabled, banks won’t have to change much – market makers will still hold accounts at their banks and they'll establish relationships with the other banks and market makers with whom they want to transact. ILP enables transactions across any number of different ledgers and exchanges using cryptographic escrow and a specialized two-phase commit protocol.
+
+Risks
+===============
+**Note** below is a discussion of HTLC risks in general, not risks inherent in implementing this BSIP.
```

Comment:
This is duplicated from elsewhere, right? Please remove and leave only those risks that are specific to this proposal (if any).

Reply:
Yes, it's duplicated from BSIP: 64, It has same risks of HASH160 algorithm.